lib/data/queries/btors/getBtorsGroupedByCountryByDepartment.ts

```diff
@@ -0,0 +1,29 @@
+import { group } from "d3";
+import prisma from "../../../../prisma/client";
+
+type QueryRow = {
+  country: string;
+  departmentId: string;
+  count: number;
+};
+
+export default async function getBtorsGroupedByCountryByDepartment() {
+  const rows: QueryRow[] = await prisma.$queryRaw`
+  SELECT
+	  "Country"."isoAlpha3" as "country",
+	  "Btor"."departmentId",
+	  CAST(COUNT(*) as INT)
+  FROM "Btor"
+  INNER JOIN "_BtorToCountry" ON "_BtorToCountry"."A" = "Btor".id
+  INNER JOIN "Country" ON "_BtorToCountry"."B" = "Country".id
+  GROUP BY "Btor"."departmentId", "Country"."isoAlpha3"
+  ORDER BY "Country"."isoAlpha3"
+  `;
+
+  const rowMap = group(rows, (d) => d.country);
```

Comment:
@Pei-Nie Mayve I am mistaken but I think grouping the data by country makes everything much more complex here? What about grouping it by department? I think this would safe some data transformation later on!

Reply:
Ok! very good suggestion! Thanks!